sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_utils.py

```diff
@@ -0,0 +1,196 @@
+                    )
+                elif SpanAttributes.NET_PEER_IP in attributes:
+                    peer_ip = attributes[SpanAttributes.NET_PEER_IP]
+                    url = "{}://{}:{}{}".format(
+                        scheme,
+                        peer_ip,
+                        peer_port,
+                        http_target,
+                    )
+    return str(url)
+
+
+def _get_target_and_path_for_http_dependency(target: Optional[str], url: Optional[str], scheme: Optional[str], attributes: Attributes) -> Tuple[Optional[str], str]:
+    target_from_url = None
+    path = ""
+    if attributes:
+        if url:
+            try:
+                parse_url = urlparse(url)
+                path = parse_url.path
+                if not path:
+                    path = "/"
+                if parse_url.port and parse_url.port == _get_default_port_http(scheme):
+                    target_from_url = parse_url.hostname
+                else:
+                    target_from_url = parse_url.netloc
+            except Exception:  # pylint: disable=broad-except
+                pass
+        if SpanAttributes.PEER_SERVICE not in attributes:
+            if SpanAttributes.HTTP_HOST in attributes:
+                host = attributes[SpanAttributes.HTTP_HOST]
+                try:
+                    # urlparse insists on absolute URLs starting with "//"
+                    # This logic assumes host does not include a "//"
+                    host_name = urlparse("//" + str(host))
+                    if host_name.port == _get_default_port_http(scheme):
+                        target = host_name.hostname
+                    else:
+                        target = str(host)
+                except Exception:  # pylint: disable=broad-except
+                    pass
+            elif target_from_url:
+                target = target_from_url
+    return (target, path)
+
+
+def _get_target_for_db_dependency(target: Optional[str], db_system: Optional[str], attributes: Attributes) -> Optional[str]:
+    if attributes:
+        db_name = attributes.get(SpanAttributes.DB_NAME)
+        if db_name:
+            if target is None:
+                target = str(db_name)
+            else:
+                target = "{}|{}".format(target, db_name)
+        elif target is None:
+            target = db_system
+    return target
+
+
+def _get_target_for_messaging_dependency(target: Optional[str], attributes: Attributes) -> Optional[str]:
+    if attributes:
+        if target is None:
+            if SpanAttributes.MESSAGING_DESTINATION in attributes:
+                target = str(attributes[SpanAttributes.MESSAGING_DESTINATION])
+            elif SpanAttributes.MESSAGING_SYSTEM in attributes:
+                target = str(attributes[SpanAttributes.MESSAGING_SYSTEM])
+    return target
+
+
+def _get_target_for_rpc_dependency(target: Optional[str], attributes: Attributes) -> Optional[str]:
+    if attributes:
+        if target is None:
```

Comment:
This seems strange to me. Why pass in an object that HAS to be None instead of just returning a new object? There's a bunch of target checks in the original, so this makes it cleaner by moving it to utils. But I feel like there's got to be a less confusing way in the future.

NIT

Reply:
It's a bit of a confusing logic but if you take a look at common schema mappings we only use the get_target_for_X_dependency logic if `target` is already not populated (from `peer.*` attributes at the top), hence we must check for `None` value before applying the logic.